compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -462,6 +462,34 @@ object TypeOps:
           else
             range(defn.NothingType, defn.AnyType)
         }
+
+      /** Deviation from standard tryWiden:
+       *  - Don't widen F-bounds
```

Comment:
Maybe the standard tryWiden shouldn't widen f-bounds either?

Reply:
Yes, this is a trial to see how it works. It may incur a performance penalty. 

If it works, I think we need a more systematic approach here: possible a field in SymDenotation to cache whether a symbol is F-bounded. The field can be set in `checkNonCyclic`, so we can avoid the computation here.

Reply:
F-bounded situations are already captured by LazyRefs. Maybe just add a case for LazyRefs to TypeOps.avoid?